orix/quaternion/quaternion.py

```diff
@@ -616,6 +674,209 @@ def identity(cls, shape: Union[int, tuple] = (1,)) -> Quaternion:
+
+        if np.sum(chi != 0) > 0:
+            not_zero = ~np.isclose(chi, 0)
+            alpha = np.arctan2(
+                np.divide(
+                    b * d - a * c, chi, where=not_zero, out=np.full_like(chi, np.inf)
+                ),
+                np.divide(
+                    -a * b - c * d, chi, where=not_zero, out=np.full_like(chi, np.inf)
+                ),
+            )
+            beta = np.arctan2(2 * chi, q03 - q12)
+            gamma = np.arctan2(
+                np.divide(
+                    a * c + b * d, chi, where=not_zero, out=np.full_like(chi, np.inf)
+                ),
+                np.divide(
+                    c * d - a * b, chi, where=not_zero, out=np.full_like(chi, np.inf)
+                ),
+            )
+            eu[..., 0] = np.where(not_zero, alpha, eu[..., 0])
+            eu[..., 1] = np.where(not_zero, beta, eu[..., 1])
+            eu[..., 2] = np.where(not_zero, gamma, eu[..., 2])
+
+        # Reduce Euler angles to definition range
+        eu[np.abs(eu) < _FLOAT_EPS] = 0
+        eu = np.where(eu < 0, np.mod(eu + 2 * np.pi, (2 * np.pi, np.pi, 2 * np.pi)), eu)
+
+        if degrees:
+            eu = np.rad2deg(eu)
+
+        return eu
+
+    def to_matrix(self) -> np.ndarray:
+        """Return the normalized quaternions as orientation matrices
+        :cite:`rowenhorst2015consistent`.
+
+        Returns
+        -------
+        om
+            Array of orientation matrices after normalizing the
+            quaternions.
+
+        Examples
+        --------
+        >>> from orix.quaternion import Quaternion
+        >>> q1 = Quaternion([[1, 0, 0, 0], [2, 0, 0, 0]])
+        >>> np.allclose(q1.to_matrix(), np.eye(3))
+        True
+        >>> q2 = Quaternion([[0, 1, 0, 0], [0, 2, 0, 0]])
+        >>> np.allclose(q2.to_matrix(), np.diag([1, -1, -1]))
+        True
+        """
+        q = self.unit
+        a, b, c, d = q.a, q.b, q.c, q.d
+        om = np.zeros(self.shape + (3, 3))
+
+        bb = b**2
+        cc = c**2
+        dd = d**2
+        qq = a**2 - (bb + cc + dd)
+        bc = b * c
+        ad = a * d
+        bd = b * d
+        ac = a * c
+        cd = c * d
+        ab = a * b
+        om[..., 0, 0] = qq + 2 * bb
+        om[..., 0, 1] = 2 * (bc - ad)
+        om[..., 0, 2] = 2 * (bd + ac)
+        om[..., 1, 0] = 2 * (bc + ad)
+        om[..., 1, 1] = qq + 2 * cc
+        om[..., 1, 2] = 2 * (cd - ab)
+        om[..., 2, 0] = 2 * (bd - ac)
+        om[..., 2, 1] = 2 * (cd + ab)
+        om[..., 2, 2] = qq + 2 * dd
+
+        return om
+
+    def to_axes_angles(self) -> Tuple[Vector3d, np.ndarray]:
+        """Return an axis-angle representation of the normalized
+        quaternions.
+        :cite:`rowenhorst2015consistent`.
+
+        Returns
+        -------
+        axis
+            an orix.vector.Vector3D object containing the axes of rotation
+        angle
+            a numpy array representing the angles of rotation
```

Comment:
Users can deduce the types from the function signature.

```suggestion
        """Return an axis-angle representation of the normalized
        quaternions :cite:`rowenhorst2015consistent`.

        Returns
        -------
        axis
            The axes of rotation.
        angle
            The angles of rotation.
```